solvation_analysis/rdf_parser.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+from scipy.interpolate import interp1d, UnivariateSpline
+import scipy
+import matplotlib.pyplot as plt
+import warnings
+
+
+def interpolate_rdf(bins, rdf, floor=0.05, cutoff=5):
+    start = np.argmax(rdf > floor)  # will return first index > rdf
+    end = np.argmax(bins > cutoff)  # will return first index > cutoff
+    bounds = (bins[start], bins[end - 1])
+    f = UnivariateSpline(bins[start:end], rdf[start:end], k=4, s=0)
+    return f, bounds
+
+
+def identify_minima(f):
+    assert isinstance(
+        f, scipy.interpolate.fitpack2.InterpolatedUnivariateSpline
+    ), "this function is compatible with the scipy.interpolate package"
+    cr_pts = f.derivative().roots()
+    cr_vals = f(cr_pts)
+    return cr_pts, cr_vals
+
+
+def plot_interpolation_fit(bins, rdf, **kwargs):
```

Comment:
For a plotting function, I'd rather it accept an "ax" argument which is a Matplotlib axis, then it works on that and returns it. Plt.plot works in mysterious ways I'd rather avoid.

Reply:
Good call. I will update to use fig, ax.

Reply:
Agreed, I personally loath matplotlibs stateful "plt.plot" interface but I was rushing.